sherpa-ncnn/csrc/recognizer.cc

```diff
@@ -124,23 +128,103 @@ class Recognizer::Impl {
     return stream;
   }
 
+  // std::unique_ptr<Stream> CreateStream(const std::vector<std::vector<int32_t>>& contexts) const {
+  std::unique_ptr<Stream> CreateStream(const char* contexts) const {
+    std::vector<std::vector<int32_t>> hotwords;
+    std::vector<int32_t> tmp;
+    std::unordered_map<std::string, int> dictionary;
+    std::ifstream file(config_.model_config.tokens);
+    if (!file) {
+        std::cerr << "open file failed: " << config_.model_config.tokens << std::endl;
+        return nullptr;
+    }
+    std::string line;
+    while (std::getline(file, line)) {
+      std::istringstream iss(line);
+      std::string first;
+      std::string second;
+      int number = 0;
+      std::size_t spacePos = line.find(' ');
+      if (spacePos != std::string::npos) {
+          first = line.substr(0, spacePos);
+          second = line.substr(spacePos + 1);
+      } else {
+          std::cerr<<"token is error"<<std::endl;
+          first = line;
+          second = "";
+		  return nullptr;
+      }
+      dictionary[first] = std::stoi(second);
+    }
+    file.close();
+	
+	std::string hotwordsfile(contexts);
```

Comment:
Could you describe the format of `hotwordsfile`?

Reply:
done